components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/resources/publisher-api.yaml

```diff
@@ -2924,6 +2924,77 @@ paths:
           schema:
             $ref: '#/definitions/Error'
 
+  /threat-protection/policies:
+    get:
+      summary: Get All Threat Protection Policies
+      description: This can be used to get all defined threat protection policies
+      responses:
+        200:
+          description: Ok. List of policies is returned
+          schema:
+            $ref: '#/definitions/ThreatProtectionPolicyList'
+
+  /threat-protection/add-policy/{apiId}/{policyId}:
+    security:
+      - OAuth2Security:
+        - apim:api_update
+    post:
+      summary: Associate a policy to an API
+      description: This can be used to asscociate a policy with an API
+      parameters:
+        - $ref: '#/parameters/apiId'
+        - $ref: '#/parameters/threatProtectionPolicyId'
+      responses:
+        200:
+          description: Ok. Successfully associated the policy with API
+        500:
+          description: Error accociating policy with API
+
+  /threat-protection/remove-policy/{apiId}/{policyId}:
```

Comment:
URI does not seem to be in correct format. HTTP DELETE can be used to delete a resource instead of using remove-* in resource paths.

Reply:
Changed and hope it is okay now.